src/website/shared/listeners/cache_suggestions.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from itertools import chain
+from typing import Any
+
+import pgpubsub
+
+from shared.channels import CVEDerivationClusterProposalChannel
+from shared.models.cached import CachedSuggestions
+from shared.models.cve import Severity
+from shared.models.linkage import CVEDerivationClusterProposal
+
+logger = logging.getLogger(__name__)
+
+
+def to_dict(instance: Any) -> dict[str, Any]:
+    opts = instance._meta
+    data = {}
+    for f in chain(opts.concrete_fields, opts.private_fields):
+        if getattr(f, "foreign_related_fields", None) is not None:
+            raw_value = getattr(instance, f.name)
+            if raw_value is not None:
+                data[f.name] = to_dict(raw_value)
+            else:
+                data[f.name] = None
+        else:
+            data[f.name] = f.value_from_object(instance)
+    for f in opts.many_to_many:
+        data[f.name] = [to_dict(i) for i in f.value_from_object(instance)]
+    return data
+
+
+def cache_new_suggestions(suggestion: CVEDerivationClusterProposal) -> None:
+    # Pre-conditions:
+    # - do we have any package_name attached?
+    if (
+        suggestion.cve.container.filter(affected__package_name__isnull=False).count()
+        == 0
+    ):
+        return
+
+    relevant_data = (
+        suggestion.cve.container.prefetch_related("affected", "metrics", "descriptions")
+        .values(
+            "title",
+            "affected__package_name",
+            "metrics__base_severity",
+            "descriptions__value",
+        )
+        .all()
+    )
+
+    relevant_piece = [x for x in relevant_data if "affected__package_name" in x]
+    if not relevant_piece:
+        # No package name.
+        return
+    relevant_piece = relevant_piece[0]
```

Comment:
@erictapen I guess we're already using more than just the first random package name (at least in some PR)?